packages/shared_preferences/shared_preferences_android/lib/shared_preferences_android.dart

```diff
@@ -4,99 +4,194 @@
 
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
-import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'package:shared_preferences_platform_interface/shared_preferences_async_platform_interface.dart';
 import 'package:shared_preferences_platform_interface/types.dart';
 
-import 'src/messages.g.dart';
+import 'legacy_shared_preferences_android.dart';
```

Comment:
Oops, I missed this in all of the implementations: you need to export this file, otherwise it's a breaking change for the package.

Let's instead do the more common thing we do with multiple files that get exported: make the top-level file just an export file, then put the actual implementations in src. See `url_launcher/lib/url_launcher.dart` for example.

Reply:
Wait, but you also renamed the Dart classes, which is also breaking. So you'll have to undo the name changes, or major-version-bump all of the implementation packages.

Reply:
the "file must be exported or else it's a breaking change" issue could probably use a test at some point